packages/vscode-extension/src/commands/Deploy.ts

```diff
@@ -1,44 +1,40 @@
 import SuiteCloudRunner from '../core/SuiteCloudRunner';
-import CommandsMetadataSingleton from '../service/CommandsMetadataSingleton';
-import { getRootProjectFolder, unwrapExceptionMessage, operationResultStatus } from '../util/ExtensionUtil';
-import { scloudOutput } from '../extension';
-import { MessageService } from '../service/MessageService';
+import MessageService from '../service/MessageService';
+import { unwrapExceptionMessage } from '../util/ExtensionUtil';
+import BaseAction from './BaseAction';
+import VSCommandOutputHandler from '../service/VSCommandOutputHandler';
 
 
-export default async function deploy() {
+export default class Deploy extends BaseAction {
+    readonly commandName: string = "project:deploy";
+    vsCommandOutputHandler = new VSCommandOutputHandler();
 
-    const deployMessageService = new MessageService('deploy');
-    const executionPath = getRootProjectFolder();
-
-    if (executionPath) {
-        const suiteCloudRunner = new SuiteCloudRunner(executionPath, CommandsMetadataSingleton.getInstance().getMetadata());
-
-        deployMessageService.showTriggeredActionInfo();
-        let deployResult;
-        try {
-            deployResult = await suiteCloudRunner.run({
-                commandName: 'project:deploy',
-                arguments: {}
-            });
-        } catch (error) {
-            deployMessageService.showErrorMessage(unwrapExceptionMessage(error));
-            return;
-        }
+    async execute(opts: {
+        suiteCloudRunner: SuiteCloudRunner,
+        messageService: MessageService
+    }) {
+        opts.messageService.showTriggeredActionInfo();
+        if (opts.suiteCloudRunner && opts.messageService) {
+            try {
+                let result = await opts.suiteCloudRunner.run({
+                    commandName: this.commandName,
+                    arguments: {}
+                });
+                if (result.status === "SUCCESS") {
```

Comment:
You can now import OperationResultStatus and change this for OperationResultStatus.SUCCESS